src/ray/object_manager/object_manager.cc

```diff
@@ -341,6 +343,12 @@ void ObjectManager::Push(const ObjectID &object_id, const NodeID &node_id) {
   if (local_objects_.count(object_id) == 0) {
     // Avoid setting duplicated timer for the same object and node pair.
     auto &nodes = unfulfilled_push_requests_[object_id];
+    // Issue a restore request if the object is on local disk. This is only relevant
+    // if the local filesystem storage type is being used.
+    auto object_url = get_spilled_object_url_(object_id);
+    if (!object_url.empty()) {
+      restore_spilled_object_(object_id, object_url, nullptr);
+    }
```

Comment:
Instead of restoring the object directly here, I think that we should issue a local pull through the pull manager so that the restoration is done under the pull manager's admission control, if possible.

Reply:
I'm going to punt this for later, since https://github.com/ray-project/ray/pull/14817 will mitigate the issue and this isn't a regression.

In the long run, direct streaming from disk means we shouldn't need admission control here.